tests/test_storageclass_deletion.py

```diff
@@ -0,0 +1,81 @@
+"""
+Automates the test OCS-297:-
+1) Create a Storage Class
+2) Create a PVC
+3) Delete the corresponding storage class
+"""
+import logging
+import pytest
+from ocsci.testlib import tier1, ManageTest
+from ocs import constants
+from tests import helpers
+from ocs_ci.utility import templating
+from ocs_ci.ocs.resources.pvc import PVC
+from ocs_ci.ocs.resources.ocs import OCS
+
+
+log = logging.getLogger(__name__)
+SC_OBJ = None
+
+
+@pytest.fixture(scope='class')
+def test_fixture(request):
+    """
+    Deleting a Storage class
+    """
+    self = request.node.cls
+
+    def finalizer():
+        teardown(self)
+    request.addfinalizer(finalizer)
+    setup(self)
+
+
+def teardown(self):
+    """
+    Remove the resources after execution of tests
+    """
+    log.info("Deleting the pvc")
+    self.pvc.delete()
+
+
+def setup(self):
+    """
+    Setting up the Environment : Creating project
+    """
+
+
+@tier1
+@pytest.mark.usefixtures(
+    test_fixture.__name__,
+)
+class TestStorageClass(ManageTest):
+
+    def test_storage_class(self):
+        """
+        Test to validate OCS-297
+        """
+        # Create a storage class
+        log.info("Creating a Storage Class")
+        self.sc_data = templating.load_yaml_to_dict(
+            constants.CSI_RBD_STORAGECLASS_YAML
```

Comment:
please use helpers.create_storage_class()

Reply:
Seems like the purpose of this test case is to delete the storage class. Therefore, everything which is done before that, as a preparation for the test itself, storage class and pvc creation, should be considered as test setup and moved to fixtures

Reply:
Done